src/Symfony/Component/Process/Pipes/AbstractPipes.php

```diff
@@ -124,16 +134,19 @@ protected function write()
                         return array($this->pipes[0]);
                     }
                 }
-                if (!isset($data[0]) && feof($input)) {
-                    // no more data to read on input resource
-                    // use an empty buffer in the next reads
-                    $this->input = null;
+                if (feof($input)) {
+                    if ($this->input instanceof \Iterator) {
+                        $this->input->next();
+                    } else {
+                        $this->input = null;
+                    }
                 }
             }
         }
 
         // no input to read on resource, buffer is empty
-        if (null === $this->input && !isset($this->inputBuffer[0])) {
+        if (!isset($this->inputBuffer[0]) && !($this->input instanceof \Iterator ? $this->input->valid() : $this->input)) {
```

Comment:
Why do we use `isset()` to test for the empty string instead of comparing it with the empty string like we usually do?